src/arch/xtensa/include/arch/task.h

```diff
@@ -107,19 +107,21 @@ static inline void task_set_data(struct task *task)
 	switch (task->priority) {
 	case TASK_PRI_MED + 1 ... TASK_PRI_LOW:
 		irq_task = *task_irq_low_get();
-		dst = &irq_task->list;
 		break;
 	case TASK_PRI_HIGH ... TASK_PRI_MED - 1:
 		irq_task = *task_irq_high_get();
-		dst = &irq_task->list;
 		break;
 	case TASK_PRI_MED:
 	default:
+#if !defined(CONFIG_HASWELL) && !defined(CONFIG_BROADWELL)
```

Comment:
ok, this is better, could we change this to CONFIG_TASK_HAVE_PRIORITY_MEDIUM in configure.ac

Reply:
ok, I will try it